roles/masquerade_external/tasks/main.yml

```diff
@@ -0,0 +1,42 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# - name: Ensure forwarding rules
+#   become: true
+#   ansible.builtin.iptables:
+#   loop:
+
+- name: Ensure masquerading rules in cifmw post routing chain
+  become: true
+  ansible.builtin.iptables:
```

Comment:
Would it be possible to do the same using firewalld module instead? I'd rather try to converge to ONE tool to manage the firewall.

I see about 2 mentions of actual iptables, vs "many more" for firewalld. You might need some rich_rule though... Care to check?

Reply:
libvirt itself uses iptables and inserts chains LIBVIRT_FWI, LIBVIRT_FWO, LIBVIRT_FWX, LIBVIRT_PRT etc.
It is important that these rules end up before any of those - I don't see any way to add chains in https://firewalld.org/documentation/man-pages/firewalld.richlanguage.html - we'd need https://firewalld.org/documentation/man-pages/firewalld.direct.html for that - which essentially just makes firewalld run iptables commands afict?

since firewalld ansible role does not support direct, I don't think we can switch.

Reply:
erf... 2024, still relying on iptables - not even nftables.... great.